src/live_restore/live_restore_state.c

```diff
@@ -0,0 +1,362 @@
+/*-
+ * Copyright (c) 2014-present MongoDB, Inc.
+ * Copyright (c) 2008-2014 WiredTiger, Inc.
+ *  All rights reserved.
+ *
+ * See the file LICENSE for redistribution information.
+ */
+
+#include "wt_internal.h"
+#include "live_restore_private.h"
+
+/*
+ * __live_restore_state_to_string --
+ *     Convert a live restore state to its string representation.
+ */
+static void
+__live_restore_state_to_string(WTI_LIVE_RESTORE_STATE state, char *state_strp)
+{
+    switch (state) {
+    case WTI_LIVE_RESTORE_STATE_NONE:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_NONE");
+        break;
+    case WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION");
+        break;
+    case WTI_LIVE_RESTORE_STATE_CLEAN_UP:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_CLEAN_UP");
+        break;
+    case WTI_LIVE_RESTORE_STATE_COMPLETE:
+        strcpy(state_strp, "WTI_LIVE_RESTORE_STATE_COMPLETE");
+        break;
+    }
+}
+
+/*
+ * __live_restore_state_from_string --
+ *     Convert a string to its live restore state.
+ */
+static int
+__live_restore_state_from_string(
+  WT_SESSION_IMPL *session, char *state_str, WTI_LIVE_RESTORE_STATE *statep)
+{
+    if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_NONE") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_NONE;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_BACKGROUND_MIGRATION;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_CLEAN_UP") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_CLEAN_UP;
+    else if (strcmp(state_str, "WTI_LIVE_RESTORE_STATE_COMPLETE") == 0)
+        *statep = WTI_LIVE_RESTORE_STATE_COMPLETE;
+    else
+        WT_RET_MSG(session, EINVAL, "Invalid state string: '%s' ", state_str);
+
+    return (0);
+}
+
+/*
+ * __live_restore_get_state_from_file --
+ *     Read the live restore state from the turtle file. If it doesn't exist we return NONE. The
+ *     caller must already hold the live restore state lock. This function takes a non-live restore
+ *     file system as it can be called in non-live restore modes.
+ */
+static int
+__live_restore_get_state_from_file(
+  WT_SESSION_IMPL *session, WT_FILE_SYSTEM *fs, WTI_LIVE_RESTORE_STATE *statep)
+{
+    WT_DECL_RET;
+
+    bool turtle_in_dest = false;
+    WTI_LIVE_RESTORE_STATE state_from_file = WTI_LIVE_RESTORE_STATE_NONE;
+    char *lr_metadata = NULL;
+
+    WT_DECL_ITEM(dest_turt_path);
+    WT_RET(__wt_scr_alloc(session, 0, &dest_turt_path));
+    WT_ERR(__wt_filename_construct(
+      session, S2C(session)->home, WT_METADATA_TURTLE, UINTMAX_MAX, UINT32_MAX, dest_turt_path));
+
+    WT_ERR(
+      fs->fs_exist(fs, (WT_SESSION *)session, (char *)(dest_turt_path)->data, &turtle_in_dest));
+
+    if (!turtle_in_dest)
+        state_from_file = WTI_LIVE_RESTORE_STATE_NONE;
```

Comment:
You could remove the `if (!turtle_in_dest)` and only have `if (turtle_in_dest)` given that the initialized value is already `NONE`. It would be good to add a comment though explaining that that is expected.

Reply:
https://github.com/wiredtiger/wiredtiger/pull/11590/commits/83b5714c773166ecf7698d9d65a82e6f0a41c6d7